### DISCUSSION TITLE
Continuous variables inference

### DISCUSSION THREAD

**Discussion by bbkjunior** [pgmpy/pgmpy#1751] [Q&A]
Hello!

I am aware that Continuous variables are not fully integrated yet, still, we can define e.g. LinearGaussianCPD. Is there any hacks/ugly workaround to calculate the inference with the BN with continuous variables even in the current state of the package?

**Comment by ankurankan** [comment]
@bbkjunior A super hacky way (and computationally expensive) way could be to use a Structural Equation Model (https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/models/SEM.py) to represent a linear gaussian BN and then compute the implied covariance matrix over all the variables (https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/estimators/SEMEstimator.py#L71). This would give us a joint  Gaussian distribution over all the variables that you can use for doing inference. Let me know if something like this might be interesting, and I can try to write a code example.

Alternatively, if you do not care about the network structure you can simply fit a multivariate gaussian distribution and do inference from it.

**Reply by bbkjunior** [reply]
@ankurankan I would be grateful if you give me some easy examples for both of these hacks.

**Reply by ankurankan** [reply]
@bbkjunior Sorry for the super late reply but I have finally implemented both fitting and prediction methods for Linear Gaussian Bayesian Networks. Here's an example:
```python
# Simulate some data to learn the model from.
In [1]: from pgmpy.utils import get_example_model

In [2]: model = get_example_model('alarm')

In [3]: from pgmpy.models import LinearGaussianBayesianNetwork

In [4]: cont_model = LinearGaussianBayesianNetwork(model.edges())

In [5]: cont_model.add_cpds(*cont_model.get_random_cpds())

In [6]: df = cont_model.simulate(int(1e3))

# Learn the model structure
In [7]: from pgmpy.estimators import PC

In [8]: dag = PC(df).estimate(ci_test='pearsonr')
Working for n conditional variables: 5: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:01<00:00,  4.08it/s]

In [10]: learnt_model = LinearGaussianBayesianNetwork(dag.edges())

# Learn the model parameters
In [11]: learnt_model.fit(df)

In [12]: learnt_model.cpds[:3]
Out[12]: 
[<LinearGaussianCPD: P(VENTALV) = N(-1.154; 0.93) at 0x76060ca380d0,
 <LinearGaussianCPD: P(PVSAT | VENTALV, FIO2) = N(2.127*VENTALV + -0.793*FIO2 + 1.613; 0.447) at 0x76060c695310,
 <LinearGaussianCPD: P(VENTLUNG | VENTALV) = N(-0.558*VENTALV + 0.579; 2.197) at 0x76060c5a7ad0]

In [22]: df_predict = df.drop(['HISTORY', 'CO'], axis=1)

In [23]: variable_order, means, var = learnt_model.predict(df_predict)

In [24]: variable_order
Out[24]: ['HISTORY', 'CO']

In [25]: means[:10]
Out[25]: 
array([[ 0.91564792, -0.13196529],
       [ 1.2164789 ,  0.08920048],
       [ 1.3937289 ,  1.46368693],
       [ 0.94240702,  1.23340451],
       [ 0.87214677, -0.09239507],
       [ 0.81074471,  0.02311998],
       [ 1.1049116 ,  3.20956097],
       [ 0.92254903, -0.97970011],
       [ 0.75519481, -0.03769225],
       [ 0.52664497, -2.59172503]])

In [26]: var
Out[26]: 
array([[2.00483033, 3.63297504],
       [2.0790856 , 0.51102163]])

```
